### PR TITLE
add coinbase sample

### PR DESCRIPTION
**Pull request by gangtao**
this PR add a sample case to show how to ingest websocket data from coinbase into proton by leveraging benthos pipeline

### CODE REVIEW

File: examples/coinbase/README.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# Demo for Benthos data pipeline and Coinbase websocket data
+
+
+
+This docker compose file demonstrates how to ingest websocket data into proton by using benthos pipeline.
```
Comment (jovezhong) [inline examples/coinbase/README.md]: WebSocket, Proton, Benthos

File: examples/coinbase/README.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            root.data = this.map_each( row -> root.columns.map_each( key -> row.get(key)) )
+
+```
+
+this pipeline will read data from coinbase websocket and then send the result to proton ingest api in a batch
+
+
+## Query you crypto price data with SQL
+
+now you can run following query to get the OHLC of the crypto data
```
Comment (jovezhong) [inline examples/coinbase/README.md]: adding :

File: examples/coinbase/README.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+SELECT
+  window_start, product_id, earliest(price) AS open, max(price) AS high, min(price) AS low, latest(price) AS close, latest(volume_24h) as acc_vol
+FROM
+  tumble(tickers, 60s)
+WHERE
+  product_id != ''
+GROUP BY
+  window_start, product_id
+SETTINGS
+  seek_to = 'earliest'
```
Comment (jovezhong) [inline examples/coinbase/README.md]: seek_to is depreciated. Please add _tp_time>earliest_ts() in WHERE

File: examples/coinbase/ddl.sql [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+SELECT
+  window_start, product_id, earliest(price) AS open, max(price) AS high, min(price) AS low, latest(price) AS close, latest(volume_24h) as acc_vol
+FROM
+  tumble(tickers, 60s)
+WHERE
+  product_id != ''
+GROUP BY
+  window_start, product_id
+SETTINGS
+  seek_to = 'earliest'
```
Comment (jovezhong) [inline examples/coinbase/ddl.sql]: depreciated

File: examples/coinbase/ddl.sql [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    SELECT
+      window_start, product_id, earliest(price) AS open, max(price) AS high, min(price) AS low, latest(price) AS close, latest(volume_24h) AS acc_vol
+    FROM
+      tumble(tickers, 300s)
+    WHERE
+      product_id != ''
+    GROUP BY
+      window_start, product_id
+    SETTINGS
+      seek_to = 'earliest'
```
Comment (jovezhong) [inline examples/coinbase/ddl.sql]: depreciated

File: examples/coinbase/ddl.sql [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+WITH ohlc AS
+  (
+    SELECT
+      window_start, product_id, earliest(price) AS open, max(price) AS high, min(price) AS low, latest(price) AS close, latest(volume_24h) AS acc_vol
+    FROM
+      tumble(tickers, 300s)
+    WHERE
+      product_id != ''
+    GROUP BY
+      window_start, product_id
```
Comment (jovezhong) [inline examples/coinbase/ddl.sql]: It's odd to use both GROUP BY and PARTITION BY for the same key

File: examples/coinbase/pipeline.yaml [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+output:
+  http_client:
+    url: http://proton:8123/proton/v1/ingest/streams/tickers
+    verb: POST
+    headers: 
+      Content-Type: application/json
+    batching:
+      count: 10
+      period: 1000ms
+      processors:
```
Comment (jovezhong) [inline examples/coinbase/pipeline.yaml:17]: Gimi will enhance the Proton ingest REST API. After that is available, this processor can be removed.

**Review by jovezhong — CHANGES_REQUESTED** [review]

File: examples/coinbase/README.md (hunk 3 above)
Comment (gangtao) [reply examples/coinbase/README.md]: tp_time has problem in recent proton

File: examples/coinbase/pipeline.yaml (hunk 7 above)
Comment (gangtao) [reply examples/coinbase/pipeline.yaml:17]: this is OK as the default api always works

**Review by jovezhong — APPROVED** [review]

**Comment by jovezhong** [comment]
(Jove Github Bot) added it to the current sprint.

**Comment by jovezhong** [comment]
(Jove Github Bot) moved this ticket out of the GitHub project(up to 1200 tickets for one project).

**Pull request merged**